src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesSanitizers.java

```diff
@@ -0,0 +1,155 @@
+      ".travis.yml", "Configure", "CMakeLists.txt"
+  };
+
+  /**
+   * A list of well-known files extensions of build configs.
+   */
+  private static final String[] BUILD_CONFIG_SUFFIXES = {
+      ".ac", ".cmake"
+  };
+
+  /**
+   * Initializes a data provider.
+   *
+   * @param fetcher An interface to GitHub.
+   */
+  public UsesSanitizers(GitHubDataFetcher fetcher) {
+    super(fetcher);
+  }
+
+  @Override
+  protected Set<Feature> supportedFeatures() {
+    return setOf(USES_ADDRESS_SANITIZER, USES_MEMORY_SANITIZER, USES_UNDEFINED_BEHAVIOR_SANITIZER);
+  }
+
+  @Override
+  protected ValueSet fetchValuesFor(GitHubProject project) throws IOException {
+    Objects.requireNonNull(project, "Oh no! Project is null!");
+
+    logger.info("Figuring out if the project uses sanitizers ...");
+
+    ValueSet values = new ValueHashSet();
+    values.update(USES_ADDRESS_SANITIZER.value(false));
+    values.update(USES_MEMORY_SANITIZER.value(false));
+    values.update(USES_UNDEFINED_BEHAVIOR_SANITIZER.value(false));
+
+    LocalRepository repository = fetcher.localRepositoryFor(project);
+
+    List<Path> files = repository.files(
+        path -> Files.isRegularFile(path) && maybeBuildConfig(path));
+
+    for (Path path : files) {
+      Optional<String> content = repository.file(path);
+      if (!content.isPresent()) {
+        continue;
+      }
+
+      List<String> sanitizers = lookForSanitizers(content.get());
+      for (String sanitizer : sanitizers) {
+        if (sanitizer.contains("address")) {
+          values.update(USES_ADDRESS_SANITIZER.value(true));
+        }
+
+        if (sanitizer.contains("memory")) {
+          values.update(USES_MEMORY_SANITIZER.value(true));
+        }
+
+        if (sanitizer.contains("undefined")) {
+          values.update(USES_UNDEFINED_BEHAVIOR_SANITIZER.value(true));
+        }
+      }
+    }
+
+    return values;
+  }
+
+  /**
+   * Returns true if a file looks like a build config, false otherwise.
+   */
+  static boolean maybeBuildConfig(Path path) {
+    String filename = path.getFileName().toString();
+
+    for (String config : BUILD_CONFIGS) {
+      if (config.equals(filename)) {
+        return true;
+      }
+    }
+
+    for (String suffix : BUILD_CONFIG_SUFFIXES) {
+      if (filename.endsWith(suffix)) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  /**
+   * Looks for options that enable one of the sanitizers.
+   *
+   * @param content The content to be checked.
+   * @return A list of lines that contain options that enable the sanitizers.
+   * @throws IOException If something went wrong.
+   */
+  static List<String> lookForSanitizers(String content) throws IOException {
+    List<String> lines = new ArrayList<>();
+
+    try (BufferedReader reader = new BufferedReader(new StringReader(content))) {
+      String line;
+      while ((line = reader.readLine()) != null) {
+        int start = line.indexOf(SANITIZER_OPTION);
```

Comment:
why don't you simply split the line based on spaces or use StringTokenizer here?

Reply:
There may be multiple compiler options in a line. The method looks for value of `-fsanitize` option. It can contain a single value such as `-fsanitize=address` or multiple values such as `-fsanitize=address,memory`. The method should be able to find both cases. It may be possible to implement this with a tokenizer but I am not sure if it would be simpler.

Reply:
Well it would look simpler if we use tokenizer or split the line based on " " because it took a while to understand for me to know why you are looking for an index of SANITIZER_OPTION. 

And you need not do this 
```
     int end = line.indexOf(" ", start);
          if (end < 0) {
            end = line.length();
          }

``` if you tokenize the line and search for the key in key=value pairs.

Reply:
Sure, we all are used to use different ways. One prefers tokenizing, another one prefers indexes. I won't argue about what is simpler. In the end, I think that neither your or mine solution works well because they don't expect a whitespace between `-fsanitize` options and they don't expect multiple `-fsanitize` options. I'll re-work the code to cover these cases.

Reply:
Thanks 👯‍♂️ :D

Reply:
I updated the code. Instead of using `StringTokenizer`, it now uses `String.split()` since `StringTokenizer` is a legacy class. Unfortunately, the procedure now is more complex but I am not sure if we can avoid that since it now covers more cases.